workspace.go

```diff
@@ -1040,6 +1040,9 @@ type WorkspaceAddTagsOptions struct {
 }
 
 func (o WorkspaceAddTagsOptions) valid() error {
+	if len(o.Tags) == 0 {
+		return ErrMissingTagIdentifier
+	}
 	for _, s := range o.Tags {
```

Comment:
If tags list is empty, ranging over an empty list will do nothing. Therefore, we need to return an error when the list is empty. Adding tags on an empty list should return an error.